src/zcbenchmarks.cpp

```diff
@@ -420,3 +445,16 @@ double benchmark_sendtoaddress(CAmount amount)
     return timer_stop(tv_start);
 }
 
+double benchmark_loadwallet()
+{
+    shutdown_wallet();
+    struct timeval tv_start;
+    bool fFirstRunRet=true;
+    timer_start(tv_start);
+    pwalletMain = new CWallet("wallet.dat");
+    DBErrors nLoadWalletRet = pwalletMain->LoadWallet(fFirstRunRet);
+    auto res = timer_stop(tv_start);
+    RegisterNodeSignals(GetNodeSignals());
+    RegisterValidationInterface(pwalletMain);
```

Comment:
When adding the re-enabling of `GenerateBitcoins()`, it would make sense to move these registrations there too, into e.g. a `restart_wallet_after_load()` method (given tha the shutdown side has been extracted.

Reply:
Done